### ISSUE TITLE
Means of programmatically overwriting pre-existing files?

### ISSUE THREAD

**Issue by abelsonlive** [wummel/patool#16]
Hi,

I'm using `patool` to programmatically unarchive a variety of archivables by calling it via `subprocess` in python.  Here's my basic code:

```python
def unarchive(file_path, file_dir):
  """
  Use patool for universal extraction.
  """
  args = ['patool', 'extract', file_path, '--outdir', file_dir]
  p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  (out, err) = p.communicate()
  return out, err
```

However, in some cases if there are duplicate files in an archive, `patool` will ask for user input as how to proceed:
```bash
EO-926055986-N81-2014/CAT. No. EO-926055986-N81-2014       14-PL-0179.xlsx already exists. Overwrite it ?
[Y]es, [N]o, [A]ll, n[E]ver, [R]ename, [Q]uit A
```

Unfortunately, this means that the `subprocess` just hangs without proceeding and `p.communicate()` never completes.  I was hoping there might be a way to add a command line flag to pass in a default response to this prompt (somewhat like `-y` for `sudo apt-get install`).

**Comment by alculquicondor** [comment]
I have encountered the same thing using patoolib. It happens that zip allows to have multiple files with the same name. I got this:

    Command `['/usr/bin/7z', 'x', '-o/tmp/tmpmRgqaa', '--', u'/srv/www_amigoserver/media/11/projects/2083/uploads/Footprints (1).zip']' returned non-zero exit status 2

Maybe the option `-y` could be added

**Comment by abelsonlive** [comment]
Any thoughts on this? It would nice to at least get the maintainers response on whether there are any potential solutions / workarounds.

**Comment by trevorprater** [comment]
From what I've seen, you have to specify arguments to bypass each use-case for every individual tool that encounters an issue like this. Here is a fix for a similar issue:

https://github.com/trevorprater/patool/commit/fddc2eb3c829f27643780293f7c8f9d96036ae57

**Comment by oliverpool** [comment]
Maybe you could do use the [`yes`](http://manpages.ubuntu.com/manpages/wily/en/man1/yes.1.html) command and a pipe: `yes | patool...`

**Comment by wummel** [comment]
We believe that the issue you reported is fixed in the source repository of patool which can be found under:
https://github.com/wummel/patool

Changelog entry:
* Added new option --non-interactive to avoid commandline prompts of
  archive programs.
  Closes: GH bug #16, #24

Thank you for reporting the issue. It is now marked as fixed. If you believe that the issue is not fixed appropriately just add a comment to this issue.

**Comment by wummel** [comment]
A new version 1.9 of patool has been released on 07.12.2015.
Therefore this bug will be closed. If you think this issue is not solved, please open a new issue.
